src/redux/actions/Dashboard/index.js

```diff
@@ -0,0 +1,38 @@
+import {
+    getDashboardTotals,
+    getDashboardChartsData,
+    getDashboardLastMessages,
+    getDashboardNewUsers,
+} from '../../services/Dashboard';
+
+export const LOADING_DASHBOARD_DATA = 'LOADING_DASHBOARD_DATA';
+export const STOP_LOADING_DASHBOARD_DATA = 'STOP_LOADING_DASHBOARD_DATA';
+export const LOAD_DASHBOARD_DATA = 'LOAD_DASHBOARD_DATA';
+
+export function fetchDashboardData() {
+    return (dispatch) => {
+        dispatch({ type: LOADING_DASHBOARD_DATA });
+        const promises = [
+            getDashboardTotals(),
+            getDashboardChartsData(),
+            getDashboardLastMessages(),
+            getDashboardNewUsers(),
+        ];
+        Promise.all(promises)
+            .then((dashboardData) => {
+                dispatch({ type: STOP_LOADING_DASHBOARD_DATA });
+
+                dispatch({
+                    type: LOAD_DASHBOARD_DATA,
```

Comment:
this is not important because this run very quickly but make more sense if you first load data an then stop the spinner